source/extensions/filters/network/zookeeper_proxy/zookeeper_utils.cc

```diff
@@ -0,0 +1,41 @@
+#include "extensions/filters/network/zookeeper_proxy/zookeeper_utils.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ZooKeeperProxy {
+
+void BufferHelper::peekInt32(Buffer::Instance& buffer, uint64_t& offset, int32_t& val) {
```

Comment:
Should all these helper take the buffer by const ref? Or are you mutating it in any way?

Reply:
Yeah good point.

Reply:
maybe not, `peekInt()` and `peekBEInt()` are not const:

https://github.com/envoyproxy/envoy/blob/master/include/envoy/buffer/buffer.h#L235